### PR TITLE
Add gift cert redemption and gift cert redemption list classes

### PR DESCRIPTION
**Pull request by SEAnderson07**
Adds GiftCertRedemption and GiftCertRedemptionList classes. Adds the gift_cert_redemption_list field to the SalesOrder class. Adds specs.

### CODE REVIEW

File: lib/netsuite/records/gift_cert_redemption.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      actions :get, :get_list, :add, :initialize, :delete, :update, :upsert, :search
+
+      fields :auth_code_amt_remaining, :auth_code_applied, :gift_cert_available
+
+      record_refs :auth_code
+
+      attr_reader :internal_id
+
+      def initialize(attributes = {})
+        initialize_from_attributes_hash(attributes)
```
Comment (iloveitaly) [inline lib/netsuite/records/gift_cert_redemption.rb]: @SEAnderson07 You need to pull the internal ID manually like this: https://github.com/NetSweet/netsuite/blob/master/lib/netsuite/records/credit_memo.rb#L36-L37

**Comment by iloveitaly** [comment]
@SEAnderson07 This looks great!! Thanks.

Can you add these records to the `basic_spec` list? https://github.com/NetSweet/netsuite/commit/fdc52bcff0db4d32a90f2b84ce358b4da6d08a6f

Ready to merge as soon as the internal ID tweak is handled.

**Comment by SEAnderson07** [comment]
@iloveitaly I made both of those updates!

**Comment by iloveitaly** [comment]
Awesome! Merged with d54c88f...bc5315a

**Pull request closed without merging**